src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -40,10 +40,22 @@ public override RefKind RefKind
         {
             get
             {
-                return
-                    ((object)_containingType == null || _containingType.TypeKind != TypeKind.Struct) ? RefKind.None :
-                    ((object)_containingMethod != null && _containingMethod.MethodKind == MethodKind.Constructor) ? RefKind.Out :
-                    RefKind.Ref;
+                if (ContainingType?.TypeKind != TypeKind.Struct)
+                {
+                    return RefKind.None;
+                }
+
+                if (_containingMethod?.MethodKind == MethodKind.Constructor)
```

Comment:
What will the value of `_containingMethod` be inside a local function which is inside a constructor?  Same for lambdas. 

Can you point me to the test which covers these cases?